effekt/shared/src/main/scala/effekt/cps/Contify.scala

```diff
@@ -85,12 +85,13 @@ object Contify {
           given Substitution = Substitution(conts = Map(k -> ContVar(k2)))
           Stmt.LetCont(id, ContLam(vparams, ks, substitute(rewrittenBody)), contify(id, rewrittenRest))
 
-        case Some(cont: ContLam) if returnsUnique && !isRecursive =>
-          val k2 = Id("k")
-          given Substitution = Substitution(conts = Map(k -> ContVar(k2)))
-          LetCont(k2, cont,
-            Stmt.LetCont(id, ContLam(vparams, ks, substitute(rewrittenBody)),
-              contify(id, rewrittenRest)))
+        // leads to `k_6 is not defined` when disabling optimizations on issue861.effekt
+        //        case Some(cont: ContLam) if returnsUnique && !isRecursive =>
+        //          val k2 = Id("k")
+        //          given Substitution = Substitution(conts = Map(k -> ContVar(k2)))
+        //          LetCont(k2, cont,
+        //            Stmt.LetCont(id, ContLam(vparams, ks, substitute(rewrittenBody)),
+        //              contify(id, rewrittenRest)))
```

Comment:
Sadly, we do not have a regression test since we need to disable optimizations.

Maybe a variant of #851 where we selectively run tests without optimizations would be useful (here JS backend so a lot of the reported segfaults etc. do not matter here.)

Reply:
In the present PR I comment this out, however this "might" have performance implications...

Reply:
I feel like we're hitting the limits of the way we currently do example-based regression tests...
We have multiple axes now, but we really only cleanly account for two of them (1 & 3):

1. by backend:
    - JS
    - LLVM
    - ...
2. by opts:
    - with opts
    - no opts
3. stdlib vs compiler:
    - stdlib
    - compiler
4. by backend-specific details:
    - with valgrind
    - without valgrind